### PR TITLE
granulate_utils: Handle unlimited cgroup quota

### PR DESCRIPTION
**Pull request by mirikl**
Take into account that a cgroup can be unlimited (`cfs_quota_us` is set to -1).

### CODE REVIEW

File: tests/granulate_utils/test_cgroups.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def wrap_read_from_control_file(mapping: Dict[str, str]) -> Callable:
+    def func(tunable_name: str) -> str:
+        return mapping[tunable_name]
+
+    return func
+
+
+@patch("granulate_utils.linux.cgroups.cpu_cgroup.BaseCgroup.__init__", return_value=None)
```
Comment (Jongy) [inline tests/granulate_utils/test_cgroups.py:14]: Maybe patch only `_verify_preconditions`? (I figure that's the reason behind this patch)

**Review by Jongy — CHANGES_REQUESTED** [review]
1 small, and in any case I'll let @michelhe decide if that's the API we want here.

File: granulate_utils/linux/cgroups/cpu_cgroup.py [hunk 2]
```diff
@@ -19,7 +19,9 @@ def set_cpu_limit_cores(self, cores: float) -> None:
 
     def get_cpu_limit_cores(self) -> float:
         period = int(self.read_from_control_file(self.cfs_period_us))
-        return int(self.read_from_control_file(self.cfs_quota_us)) / period
+        quota = int(self.read_from_control_file(self.cfs_quota_us))
+        # if quota is set to -1 it means this cgroup is unlimited
+        return quota / period if quota != -1 else -1.0
```
Comment (michelhe) [inline granulate_utils/linux/cgroups/cpu_cgroup.py:24]: It would be nicer to return `None` instead of `-1` but linux likes using `-1` for these tunables
I'm fine with keeping this as `-1` to represent unlimited

Comment (d3dave) [reply granulate_utils/linux/cgroups/cpu_cgroup.py:24]: that makes sense when you are writing in C and using the sign bit to represent a null value, but we are in python here so testing for `< 0` is more confusing than testing `is None`.

File: tests/granulate_utils/test_cgroups.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+from unittest.mock import patch
+
+from granulate_utils.linux.cgroups.cpu_cgroup import CpuCgroup
+
+
+def wrap_read_from_control_file(mapping: Dict[str, str]) -> Callable:
+    def func(tunable_name: str) -> str:
+        return mapping[tunable_name]
+
+    return func
```
Comment (d3dave) [inline tests/granulate_utils/test_cgroups.py:11]: can just do `return mapping.__getitem__`

File: granulate_utils/linux/cgroups/cpu_cgroup.py (hunk 2 above)
Comment (mirikl) [reply granulate_utils/linux/cgroups/cpu_cgroup.py:24]: @d3dave , I'd rather not change the return value to `None` as it changes the API since the function will return `Optional[Float]` instead of `float`. I don't mind changing it but I'd like to talk offline about what else I need to check for this change as I'm not entirely sure.

Comment (d3dave) [reply granulate_utils/linux/cgroups/cpu_cgroup.py:24]: Ok. No real issue with keeping it -1.0. The only two callsites I found don't test the value anyway.

**Review by d3dave — APPROVED** [review]
